source/fe/mapping_q_generic.cc

```diff
@@ -3522,8 +3521,42 @@ namespace
       }
 
       case 8:
-        Assert(false, ExcNotImplemented());
+      {
+        Assert(spacedim >= 3, ExcImpossibleInDim(spacedim));
+        unsigned int m=1;
+        for ( ; m < n; ++m)
+          if (m*m*m == n)
+            break;
+        // is n a cube number
+        Assert(m*m*m==n, ExcInternalError());
+
+        // If m points are passed, two of them are the outer vertices, and we can
+        // only compute (degree-1)*(degree-1)*(degree-1) intermediate points.
+        for (unsigned int k=0, c=0; k<m; ++k)
+          {
+            const double z=line_support_points.point(1+k)[0];
+            for (unsigned int i=0; i<m; ++i)
+              {
+                const double y=line_support_points.point(1+i)[0];
+                for (unsigned int j=0; j<m; ++j, ++c)
+                  {
+                    const double x=line_support_points.point(1+j)[0];
+
```

Comment:
I think it would be best to replace the hard-coded 3_linear functions here with a loop over `l` using `GeometryInfo<3>::	d_linear_shape_function (line_support_points.point(1+k), const unsigned int l)`